docs/source/reference/udf_decorators.rst

```diff
@@ -0,0 +1,153 @@
+User-Defined Functions
+======================
+
+This section provides an overview of how you can create and use a custom user-defined function (UDF) in your queries. For example, you could write an UDF that wraps around a PyTorch model.
+
+
+Part 1: Writing a custom UDF
+-----
+.. admonition:: Illustrative UDF implementation
+
+    During each step, use `this UDF implementation <https://github.com/georgia-tech-db/eva/blob/master/eva/udfs/decorators/yolo_object_detection_decorators.py>`_  as a reference.
+
+1. Create a new file under `udfs/` folder and give it a descriptive name. eg: `yolo_object_detection.py`. 
+
+.. note::
+
+    UDFs packaged along with EVA are located inside the `udfs <https://github.com/georgia-tech-db/eva/tree/master/eva/udfs>`_ folder.
+
+2. Create a Python class that inherits from `PytorchClassifierAbstractUDF`.
+
+* The `PytorchClassifierAbstractUDF` is a parent class that defines and implements standard methods for model inference.
+
+* The functions setup and forward should be implemented in your child class. These functions can be implemented with the help of Decorators.
+
+Setup
+-----
+
+An abstract method that must be implemented in your child class. The setup function can be used to initialize the parameters for executing the UDF. The parameters that need to be set are 
+
+- cacheable: bool
+ 
+  - True: Cache should be enabled. Cache will be automatically invalidated when the UDF changes.
+  - False: cache should not be enabled.
+- udf_type: str
+  
+  - object_detection: UDFs for object detection.
+- batchable: bool
+  
+  - True: Batching should be enabled
+  - False: Batching is disabled.
+
+The custom setup operations for the udf can be written inside the function in the child class. If there is no custom logic required, then you can just mention pass in the function definition.
+
+Example of the setup function
+
+.. code-block:: python
+
+  @setup(cachable=True, udf_type="object_detection", batchable=True)
+  def setup(self, threshold=0.85):
+      #custom setup function that is specific for the UDF
+      self.threshold = threshold 
+      self.model = torch.hub.load("ultralytics/yolov5", "yolov5s", verbose=False)
+
+Forward
```

Comment:
Should we rename this function as `Execute` since not all UDFs will do `Inference`?

Reply:
Maybe in a future PR.